scripts/04_generate_flood_polygons.py

```diff
@@ -1,81 +1,187 @@
+
     gdf_admin = stratus.codab.load_codab_from_fieldmaps(args.iso3, 0)
     bbox = gdf_admin.total_bounds
-    items = query_gfm_stac(bbox, args.end_date)
-    
-    if len(items) == 0:
-        logger.error("No STAC items found for the specified criteria")
-        return
-    
-    flood_composite, unique_dates = create_flood_composite(
-        items, bbox, args.n_latest, mode=args.flood_mode
-    )
 
-    flood_polygons = raster_to_polygons(flood_composite)
-    
+    # Use tiled processing for large countries
+    if args.use_tiling:
+        logger.info(f"Using tiled processing with {args.tile_size}° tiles")
+
+        # Note: Tiled processing doesn't support provenance yet
+        flood_polygons, unique_dates = process_country_tiled(
+            bbox=bbox,
+            end_date=args.end_date,
+            n_latest=args.n_latest,
+            n_search=args.n_search,
+            mode=args.flood_mode,
+            tile_size=args.tile_size,
+            return_stack=False
+        )
+        stack_flood_max = None  # Provenance not supported for tiled processing
+
+    else:
+        # Standard processing for small countries
+        items = query_gfm_stac(bbox, args.end_date, args.n_search)
+
+        if len(items) == 0:
+            logger.error("No STAC items found for the specified criteria")
+            return
+
+        # Create flood composite with stack for provenance
+        flood_composite, unique_dates, stack_flood_max = create_flood_composite(
+            items, bbox, args.n_latest, mode=args.flood_mode, return_stack=True
+        )
+
+        logger.info("Converting flood raster to polygons...")
+        try:
+            flood_polygons = raster_to_polygons(flood_composite)
+            logger.info(f"✅ Polygon conversion complete")
+        except Exception as e:
+            logger.error(f"❌ Polygon conversion failed: {e}")
+            raise
+
     if len(flood_polygons) == 0:
         logger.warning("No polygons created")
         return
-    
-    date_str = args.end_date.replace('-', '')
+
+    date_str = args.end_date.replace("-", "")
     filename_base = f"{args.iso3.lower()}_flood_{args.flood_mode}_{date_str}"
     output_path = args.output_dir / filename_base
-    
+
     output_path = generate_cache_key(args.iso3, unique_dates, None, args.flood_mode)
-    export_polygons(
-        flood_polygons, 
-        output_path,
-        local=False,
-        blob=True
-    )
-    
-    logger.info("=" * 60)
-    logger.info("FLOOD POLYGON GENERATION COMPLETE")
-    logger.info("=" * 60)
-    logger.info(f"   Polygons created: {len(flood_polygons)}")
+    export_polygons(flood_polygons, output_path, local=False, blob=True)
+
+    # Generate and upload provenance raster (only for non-tiled processing)
+    if stack_flood_max is not None:
+        logger.info("\n" + "=" * 60)
+        logger.info("GENERATING PROVENANCE RASTER")
+        logger.info("=" * 60)
+
+        provenance_idx, date_mapping = create_provenance_raster(
```

Comment:
This date mapping gets a little bit lost in the logs. Would it be possible to store this in the COG metadata?

Reply:
doing some playing arround where im able to store some metadata as data-array attributes which then get preserved when written to a cog via stratus. Heres the output where you can see where the metadata lives and what it looks like in qgis:

<img width="1870" height="304" alt="image" src="https://github.com/user-attachments/assets/d22212ce-aae9-4a32-ae1e-9c7b0bde5b56" />